tv_apps/smart-system/js/sync_manager.js

```diff
@@ -501,6 +487,35 @@
 
     /** Helpers **/
 
+    observeSettings: function() {
+      var setSettingObserver = setting => {
+        return new Promise(resolve => {
+          SettingsListener.observe(setting, false, enabled => {
+            this.debug('Observed ' + setting + ' changed to ' + enabled);
+            enabled ? this.settings.set(setting, enabled)
+                    : this.settings.delete(setting);
+            resolve && resolve();
+          });
+        });
+      };
+
+      var promises = [];
+      ['sync.collections.bookmarks.enabled',
+       'sync.collections.history.enabled',
+       'sync.collections.passwords.enabled',
+       'sync.collections.bookmarks.readonly',
+       'sync.collections.history.readonly',
+       'sync.collections.passwords.readonly',
+       'sync.server.url',
+       'sync.scheduler.interval',
+       'sync.scheduler.wifionly',
+       'sync.fxa.audience'].forEach(setting => {
```

Comment:
I usually do `Promise.all([ ... ].map(setting => setSettingObserver(setting)));` which avoids pushing to a second array, but I guess your syntax is easier to read, so you choose.